bravado/fido_client.py

```diff
@@ -93,14 +93,54 @@ def json(self, **_):
         return self._delegate.json()
 
 
+class FidoFutureAdapter(FutureAdapter[T]):
+    """
+    This is just a wrapper for an EventualResult object from crochet.
+    It implements the 'result' method which is needed by our HttpFuture to
+    retrieve results.
+    """
+
+    timeout_errors = (fido.exceptions.HTTPTimeoutError,)
+    connection_errors = (
+        fido.exceptions.TCPConnectionError,
+        twisted.internet.error.ConnectingCancelledError,
+        twisted.internet.error.DNSLookupError,
+        twisted.web.client.RequestNotSent,
+    )
+
+    def __init__(self, eventual_result):
+        # type: (typing.Any) -> None
+        self._eventual_result = eventual_result
+
+    def result(self, timeout=None):
+        # type: (typing.Optional[float]) -> T
+        try:
+            return self._eventual_result.wait(timeout=timeout)
+        except crochet.TimeoutError:
+            self.cancel()
+            six.reraise(
+                fido.exceptions.HTTPTimeoutError,
+                fido.exceptions.HTTPTimeoutError(
+                    'Connection was closed by fido after blocking for '
+                    'timeout={timeout} seconds waiting for the server to '
+                    'send the response'.format(timeout=timeout)
+                ),
+                sys.exc_info()[2],
+            )
+
+    def cancel(self):
+        # type: () -> None
+        self._eventual_result.cancel()
+
+
 class FidoClient(HttpClient):
     """Fido (Asynchronous) HTTP client implementation.
     """
 
     def __init__(
         self,
-        future_adapter_class=None,  # type: typing.Optional[typing.Type[FidoFutureAdapter]]
-        response_adapter_class=None,  # type: typing.Optional[typing.Type[FidoResponseAdapter]]
+        future_adapter_class=FidoFutureAdapter,  # type: typing.Optional[typing.Type[FidoFutureAdapter]]
+        response_adapter_class=FidoResponseAdapter,  # type: typing.Optional[typing.Type[FidoResponseAdapter]]
```

Comment:
Let's remove the `Optional` since `None` is no longer a value we support. Same for the RequestsClient.

Reply:
Done